lib/liquid/block_body.rb

```diff
@@ -70,49 +70,57 @@ def render(context)
       output = []
       context.resource_limits.render_score += @nodelist.length
 
-      @nodelist.each do |token|
-        # Break out if we have any unhanded interrupts.
+      idx = 0
+      while node = @nodelist[idx]
+        render_node_to_output(node, output, context)
         break if context.interrupt?
-
-        begin
-          # If we get an Interrupt that means the block must stop processing. An
-          # Interrupt is any command that stops block execution such as {% break %}
-          # or {% continue %}
-          if token.is_a?(Continue) || token.is_a?(Break)
-            context.push_interrupt(token.interrupt)
-            break
-          end
-
-          node_output = render_node(token, context)
-
-          unless token.is_a?(Block) && token.blank?
-            output << node_output
-          end
-        rescue MemoryError => e
-          raise e
-        rescue UndefinedVariable, UndefinedDropMethod, UndefinedFilter => e
-          context.handle_error(e, token.line_number)
-          output << nil
-        rescue ::StandardError => e
-          line_number = token.is_a?(String) ? nil : token.line_number
-          output << context.handle_error(e, line_number)
-        end
+        idx += 1
       end
 
       output.join
     end
 
     private
 
+    def render_node_to_output(node, output, context)
+      case node
+      when String
+        node_output = node
+        check_resources(context, node_output)
+        output << node_output
+      when Continue, Break
+        # If we get an Interrupt that means the block must stop processing. An
+        # Interrupt is any command that stops block execution such as {% break %}
+        # or {% continue %}
+        context.push_interrupt(node.interrupt)
+      when Block
+        node_output = render_node(node, context)
+        output << node_output unless node.blank?
```

Comment:
If `node.blank?` is true, did we actually have to call `render_node`? Or could we check that before?

Reply:
`render_node` may have side effects for block tags. At least the test suite lights up, if `blank?` it's checked before.